tools/shared.py

```diff
@@ -55,17 +55,22 @@
 if sys.version_info < (2, 7, 12):
   logger.debug('python versions older than 2.7.12 are known to run into outdated SSL certificate related issues, https://github.com/emscripten-core/emscripten/issues/6275')
 
+# warning about absolute-paths is disabled by default, and not enabled by -Wall
+diagnostics.add_warning('absolute-paths', enabled=False, part_of_all=False)
+diagnostics.add_warning('separate-asm')
+diagnostics.add_warning('almost-asm')
+diagnostics.add_warning('invalid-input')
+# Don't show legacy settings warnings by default
+diagnostics.add_warning('legacy-settings', enabled=False, part_of_all=False)
+# Catch-all for other emcc warnings
+diagnostics.add_warning('linkflags')
+diagnostics.add_warning('emcc')
+diagnostics.add_warning('undefined')
+diagnostics.add_warning('version-check')
```

Comment:
maybe these could be in `diagnostics.py`?

Reply:
I kind of like the specific warnings being define by emcc.   Kind of makes `diagnostics.py` more like a re-usable library.